### PR TITLE
store/tikv: move safepoint test to /tests

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
store/tikv has dependencies on `mockstore` and `unistore` because tests need to use them. We can resolve it by moving all tests to standaline package.

Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- move test files into `store/tikv/tests`

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/tests/util_test.go [hunk 1]
```diff
@@ -98,3 +100,17 @@ func (s *OneByOneSuite) TearDownSuite(c *C) {
 		withTiKVGlobalLock.RUnlock()
 	}
 }
+
+func encodeKey(prefix, s string) []byte {
```
Comment (AndreMouche) [inline store/tikv/tests/util_test.go:104]: duplicate function make me unhappy to give LGTM

Comment (disksing) [reply store/tikv/tests/util_test.go:104]: I will remove one of them in the end.

**Review by andylokandy — APPROVED** [review]

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**